File: frontend/lib/dto/NotificationPreferencesDTO.dart
class NotificationPreferencesDTO {
  bool cardStatusNotification;
  bool cardCancelNotification;
  bool newCardRequestNotification;
  bool cardReplacementNotification;
  bool travelPlanNotification;
  bool transactionNotification;

  NotificationPreferencesDTO({
    required this.cardStatusNotification,
    required this.cardCancelNotification,
    required this.newCardRequestNotification,
    required this.cardReplacementNotification,
    required this.travelPlanNotification,
    required this.transactionNotification,
  });

  factory NotificationPreferencesDTO.fromJson(Map<String, dynamic> json) {
    return NotificationPreferencesDTO(
      cardStatusNotification: json['cardStatusNotification'] ?? false,
      cardCancelNotification: json['cardCancelNotification'] ?? false,
      newCardRequestNotification: json['newCardRequestNotification'] ?? false,
      cardReplacementNotification: json['cardReplacementNotification'] ?? false,
      travelPlanNotification: json['travelPlanNotification'] ?? false,
      transactionNotification: json['transactionNotification'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'cardStatusNotification': cardStatusNotification,
      'cardCancelNotification': cardCancelNotification,
      'newCardRequestNotification': newCardRequestNotification,
      'cardReplacementNotification': cardReplacementNotification,
      'travelPlanNotification': travelPlanNotification,
      'transactionNotification': transactionNotification,
    };
  }
}

File: microservices/microservices/event-service/src/main/java/hps/ma/eventservice/dto/NotificationPreferencesDTO.java
package hps.ma.eventservice.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class NotificationPreferencesDTO {
    @JsonProperty("cardStatusNotification")
    private boolean cardStatusChanges;

    @JsonProperty("cardCancelNotification")
    private boolean cardCancelReactivate;

    @JsonProperty("newCardRequestNotification")
    private boolean newCardRequest;

    @JsonProperty("cardReplacementNotification")
    private boolean cardReplacementRequest;

    @JsonProperty("travelPlanNotification")
    private boolean travelPlanStatus;

    @JsonProperty("transactionNotification")
    private boolean transactionAlert;
}
